ops/testing.py

```diff
@@ -878,8 +878,15 @@ def remove_relation(self, relation_id: int) -> None:
         for unit_name in rel_list_map[relation_id].copy():
             self.remove_relation_unit(relation_id, unit_name)
 
+        prev_broken_id = None  # Silence linter warning.
+        if self._model is not None:
+            prev_broken_id = self._model._relations._broken_relation_id
+            self._model.relations._broken_relation_id = relation_id
```

Comment:
I'm not sure I quite follow what these changes are doing -- can you explain?

Reply:
Does the comment make it clear?

Reply:
Ah yes, thanks!